tests/integration_tests/performance/test_boottime.py

```diff
@@ -16,46 +16,117 @@
 # Regex for obtaining boot time from some string.
 TIMESTAMP_LOG_REGEX = r'Guest-boot-time\s+\=\s+(\d+)\s+us'
 
+NO_OF_MICROVMS = 10
 
-@pytest.mark.timeout(120)
-def test_microvm_boottime_no_network(test_microvm_with_boottime):
+
+def test_single_microvm_boottime_no_network(test_microvm_with_boottime):
     """Check guest boottime of microvm without network."""
-    boottime_us = _test_microvm_boottime(test_microvm_with_boottime, None)
+    log_fifo, _ = _configure_vm(test_microvm_with_boottime)
+    time.sleep(0.4)
+    boottime_us = _test_microvm_boottime(log_fifo)
     print("Boot time with no network is: " + str(boottime_us) + " us")
 
 
-def test_microvm_boottime_with_network(
-        test_microvm_with_boottime,
+@decorators.test_context('boottime', NO_OF_MICROVMS)
+def test_multiple_microvm_boottime_no_network(test_multiple_microvms):
+    """Check guest boottime without network when spawning multiple microvms."""
+    microvms = test_multiple_microvms
+    assert microvms
+    assert len(microvms) == NO_OF_MICROVMS
+    log_fifos = []
+    for i in range(NO_OF_MICROVMS):
+        log_fifo, _ = _configure_vm(microvms[i])
+        log_fifos.append(log_fifo)
+    time.sleep(0.4)
+    for i in range(NO_OF_MICROVMS):
+        boottime_us = _test_microvm_boottime(log_fifos[i])
+        print("Boot time with no network for microvm {} is: {} us".format(
```

Comment:
Should we assert on the boot time here instead? Printing it for 100 microvms will pollute the CI log with information that doesn't seem useful for the happy case. Same for the other test (test_multiple_microvm_boottime_with_network).

Reply:
_test_microvm_boottime asserts on the bootime.
```
def _test_microvm_boottime(log_fifo):
    ...
    ...
    assert boot_time_us > 0
    assert boot_time_us < MAX_BOOT_TIME_US
    return boot_time_us
```
After asserting, boot_time_us is returned for printing purposes only. Indeed, in this case we would be polluting the stdin with a lot of messages. I removed the printing part. Take a look.

Reply:
I'd be curious to see some data on this though. Maybe not at per-vm granularity, but perhaps in intervals - for instance, how many microvms booted in 125-150 ms, how many in 150-200 etc. What do you think?

Reply:
I think that by doing so I would be deflecting from the initial purpose of this PR which was to add support for spawning multiple microvms not adding performance tests when spawning multiple vms. I added this test to exemplify how we would use the multiple microvms decorator and I do not intent to invest more time in solving issues that may occur when actually validating the performance of firecracker when multiple vms are spawned.